x/bep3/types/errors.go

```diff
@@ -0,0 +1,111 @@
+	DefaultCodespace             sdk.CodespaceType = ModuleName
+	CodeInvalidLockTime          CodeType          = 1
+	CodeInvalidModulePermissions CodeType          = 2
+	CodeAtomicSwapNotFound       CodeType          = 3
+	CodeInvalidCoinDenom         CodeType          = 4
+	CodeAmountTooLarge           CodeType          = 5
+	CodeAmountTooSmall           CodeType          = 6
+	CodeAtomicSwapAlreadyExists  CodeType          = 7
+	CodeInvalidClaimSecret       CodeType          = 8
+	CodeOnlySameChain            CodeType          = 9
+	CodeOnlyOriginalCreator      CodeType          = 10
+	CodeAssetNotSupported        CodeType          = 11
+	CodeAssetNotActive           CodeType          = 12
+	CodeInvalidHeightSpan        CodeType          = 13
+	CodeAtomicSwapHasExpired     CodeType          = 14
+	CodeOnlyDeputy               CodeType          = 15
+)
+
+// ErrInvalidLockTime Error constructor
+func ErrInvalidLockTime(codespace sdk.CodespaceType) sdk.Error {
+	return sdk.NewError(codespace, CodeInvalidLockTime, fmt.Sprintf("invalid lock time: must be greater than minimum lock time"))
+}
+
+// ErrInvalidModulePermissions error for when module doesn't have valid permissions
+func ErrInvalidModulePermissions(codespace sdk.CodespaceType, permission string) sdk.Error {
+	return sdk.NewError(codespace, CodeInvalidModulePermissions, fmt.Sprintf("module does not have required permission '%s'", permission))
+}
+
+// ErrAtomicSwapNotFound error for when an atomic swap is not found
+func ErrAtomicSwapNotFound(codespace sdk.CodespaceType, id []byte) sdk.Error {
+	return sdk.NewError(codespace, CodeAtomicSwapNotFound, fmt.Sprintf("AtomicSwap %s was not found", BytesToHex(id)))
+}
+
+// ErrInvalidCoinDenom error for when coin denom doesn't match AtomicSwap coin denom
+func ErrInvalidCoinDenom(codespace sdk.CodespaceType, coinDenom string, swapCoinDenom string) sdk.Error {
+	return sdk.NewError(codespace, CodeInvalidCoinDenom, fmt.Sprintf("coin denom %s doesn't match AtomicSwap coin denom %s", coinDenom, swapCoinDenom))
+}
+
+// ErrAmountTooLarge error for when a coin amount will put the asset supply over the asset limit
+func ErrAmountTooLarge(codespace sdk.CodespaceType, coin sdk.Coin) sdk.Error {
+	return sdk.NewError(codespace, CodeAmountTooLarge, fmt.Sprintf("deposit of asset %s not allowed due to the asset's global supply limit", coin))
+}
+
+// ErrAmountTooSmall error for when a coin amount is 0
+func ErrAmountTooSmall(codespace sdk.CodespaceType, coin sdk.Coin) sdk.Error {
+	return sdk.NewError(codespace, CodeAmountTooSmall, fmt.Sprintf("coin %s amount is below the limit for this operation", coin))
+}
+
+// ErrAtomicSwapAlreadyExists error for when an AtomicSwap with this swapID already exists
+func ErrAtomicSwapAlreadyExists(codespace sdk.CodespaceType, swapID cmn.HexBytes) sdk.Error {
+	return sdk.NewError(codespace, CodeAtomicSwapAlreadyExists, fmt.Sprintf("coin %s amount is below the limit for this operation", swapID))
```

Comment:
```suggestion
	return sdk.NewError(codespace, CodeAtomicSwapAlreadyExists, fmt.Sprintf("swap %s already exists", swapID))
```

Reply:
Ah, this is why I was seeing a different error message.